autogpt_platform/backend/migrate/run_migrations.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import os
+import subprocess
+
+import psycopg2
+
+from backend.util.settings import Settings
+
+
+def run_prisma_migrations():
+    result = subprocess.run(
+        ["prisma", "migrate", "deploy"], capture_output=True, text=True
+    )
+    if result.returncode != 0:
+        logging.info("Prisma migration failed:")
+        logging.error(result.stderr)
+        return False
+    return True
+
+
+def run_custom_migrations():
+    settings = Settings()
+    db_url = settings.config.direct_database_url
+    if not db_url:
+        logging.error("No Direct DB URL Provided")
+        return
+    conn = psycopg2.connect(db_url)
+
+    try:
+        with conn.cursor() as cur:
+            custom_migrations_dir = "./custom_migrations"
+            for filename in sorted(os.listdir(custom_migrations_dir)):
+                if filename.endswith(".sql"):
+                    with open(os.path.join(custom_migrations_dir, filename), "r") as f:
+                        sql = f.read()
+                        cur.execute(sql)
+                    logging.info(f"Executed custom migration: {filename}")
+            conn.commit()
+    except Exception as e:
+        logging.exception(f"Custom migration failed: {e}")
+        conn.rollback()
+    finally:
+        conn.close()
+
+
+def main():
+    logging.info("Starting migrations")
+    if run_prisma_migrations():
+        run_custom_migrations()
+    logging.info("Migrations completed")
+
+
+if __name__ == "__main__":
+    run_prisma_migrations()
+    run_custom_migrations()
```

Comment:
Might be a silly question: can't we manually add our SQL migration using the prisma migration format into the existing migrations folder and just let `prisma migrate` run it? Why the need to manual script?

Reply:
not sure I understand the question, but what I'm trying to solve for is: 

prisma won't migrate any schemas and tables it doesn't own. Auth is owned and manged by supabase, so prisma won't do it

Reply:
And the ownership is defined by the table being defined in `schema.prisma`? what happen if we do DDL query on the migration with some table defined by ourselves? Like manually adding the SQL file into the existing migration folder.

Reply:
Prisma was not compatible with adding the sql to the files directly by hand. There’s other migrations that do that (like the vectorizing for search in market) but without this method, it doesn’t work

Reply:
This seems to be working fine for me
<img width="1203" alt="image" src="https://github.com/user-attachments/assets/6c847d3d-24e0-43aa-a705-06f1a54442ee">

```
  └─ 20241007115715_testing_new_migration/
    └─ migration.sql

✔ Generated Prisma Client Python (v0.13.1) to ./.venv/lib/python3.12/site-packages/prisma in 109m
```

What was the incompatibility?

Reply:
For an equivalent test, make that table under the auth schema

Reply:
<img width="1359" alt="image" src="https://github.com/user-attachments/assets/4776f859-c905-4562-b1ec-29d7d38b0ea9">

Seems to be working.

[[How to use Prisma ORM with multiple database schemas | Prisma Documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/multi-schema)](https://www.prisma.io/docs/orm/prisma-schema/data-model/multi-schema)

Reply:
Heck throw the trigger script in and give it a go, maybe we missed something

Reply:
What was the error? But before I'm in too deep into this rabbit hole, do we really need a stored procedure? Can this single trigger be replaced with a single TS/Python insert and we eliminate all of this added complexity?

How is the user inserted atm? is it through an API not a SQL query ?

Reply:
Two ways right now which is first in Supabase signup which is not supposed to be altered inserts it into auth and then after that the frontend makes a query to the backend on a random(not really but feels like someone just chose it) page load to get the user which causes half the queries to fail because they all fire at the same time and for the db to generate its own uuid for the user which isn’t linked to Supabase in any way

Reply:
So we can do wrap transaction & insert on the signup function?

Btw this works
<img width="1168" alt="image" src="https://github.com/user-attachments/assets/2cb3f916-fe8d-465b-a78f-6099a4923e6e">

I don't have auth.users currently locally, but adding trigger works just fine.

Reply:
It’s crazy this works for you, @aarushik93 and I both got errors :/

does it show in like postico or whatever db explorer you use that the function is actually on the correct schema

Reply:
Hmm nope, it's just a SQL query after all, why is it not working?

I initially added
```
  previewFeatures = ["multiSchema"]
```
and mentioned auth and platform as part of the schema as described here:
https://www.prisma.io/docs/orm/prisma-schema/data-model/multi-schema#how-to-include-multiple-database-schemas-in-your-prisma-schema

But it seems like it's not needed since turns out you are not querying the data through Prisma at all.

Reply:
To be honest with you, it was well over two weeks ago when I tried it. I don’t know why it didn’t work, it just didn’t. 

If you got it working in the scope of the pr, push it and let’s get it merged and then we don’t need to deal with any of the custom migration stuff. It didn’t work due to various errors for rushi or I